torch/csrc/jit/profiling_graph_executor_impl.cpp

```diff
@@ -11,29 +18,65 @@ bool& getProfilingMode() {
   return profiling_mode;
 }
 
-static std::shared_ptr<Graph> prepareGraph(const std::shared_ptr<Graph>& graph) {
+std::shared_ptr<Graph> ProfilingGraphExecutorImpl::prepareGraph(
+    const std::shared_ptr<Graph>& graph,
+    Stack& stack) {
   auto g = graph->copy();
+  ArgumentSpec spec =
+      arg_spec_creator_.create(autograd::GradMode::is_enabled(), stack);
+  arg_spec_creator_.specializeTypes(*g, spec);
   runRequiredPasses(g);
+  PropagateRequiresGrad(g);
+  if (needsGradient(g)) {
+    auto diff_nodes = CreateAutodiffSubgraphs(
+        g, getAutodiffSubgraphInlining() ? autodiffSubgraphNodeThreshold : 1);
+    for (Node* dnode : diff_nodes) {
+      auto diff_graph = std::move(dnode->g(attr::Subgraph));
+      Gradient gradient = differentiate(diff_graph);
+      // do not optimize DifferentiableGraphs, since
+      // ideally they will be profiled and then optimized separetely
+      // when their corresponding DifferentiableGraphOp is called
+      packGradient(gradient, dnode);
+    }
+    InlineAutodiffSubgraphs(
```

Comment:
We need to make sure that this pass, which undoes the creation of too-small gradients, does not accidentally lose profiling information for these things, because it is going to remove the DifferentiableGraphs where the optimization for these things would have taken place.

Reply:
I'll double check